app/models/notification.rb

```diff
@@ -209,6 +214,14 @@ def self.build_from_payload(auth_key, payload)
         n.content_available = !!payload[:content_available]
       end
 
+      if payload[:priority].present?
+        n.priority = payload[:priority]
+      end
+
+      if payload[:expiration].present?
+        n.expiration = Time.parse(payload[:expiration]) rescue nil
```

Comment:
I prefer to pass times around as unix timestamps. I don't think this app has any other methods that accept timestamps so we can set the standard by doing this now and using `Time.at(xxx)`.

The `rescue nil` isn't very nice and shouldn't be needed using `Time.at` as long as you make sure that `payload[:expiration]` is a positive integer greater than `Time.now.to_i`.